user_status.c

```diff
@@ -153,6 +153,7 @@ static const char *state_to_str(const char state) {
 
   errno = EINVAL;
   die("state_to_str:");
+  return NULL; // supress control reaches end of non void function warning
```

Comment:
A better fix would be to mark the `die` function as `noreturn`

Reply:
do you want to go `-std=c23` or non standard extension?
former would probably require other work

Reply:
`_Noreturn` works without any extensions i think

Reply:
that is still compiler specific, I'm just asking if we wanna go non-std route
Although, it doesn't matter since clang and gcc both support this attribute

Reply:
`_Noreturn` is fine, even `cproc` supports it

Reply:
just realised a few things
- marking the function `_Noreturn` will cause undefined behavior because `die()` selectively returns.
- `_Noreturn` as a keyword was introduced in C11 and deprecated in C23. `slstatus` is compiled with <C11 iirc (not completely sure)

Reply:
`die` never returns, I meant `die`, not the utility function

Changing the std to c11 is fine